ql/src/java/org/apache/hadoop/hive/ql/parse/RewriteSemanticAnalyzer.java

```diff
@@ -409,7 +409,14 @@ private boolean isTargetTable(Entity entity, Table targetTable) {
      * is this the right way to compare?  Should it just compare paths?
      * equals() impl looks heavy weight
      */
-    return targetTable.equals(entity.getTable());
+    long targetWriteId = targetTable.getTTable().getWriteId();
```

Comment:
Can you add a comment here as to why we need to ignore the writeId here?

Reply:
Done